src/ContentFileMigrationTask.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace SilverStripe\Assets;
+
+use SilverStripe\Assets\File;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Dev\BuildTask;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DB;
+use SilverStripe\Versioned\Versioned;
+
+/**
+ * Class ContentFileMigrationTask
+ * @package SilverStripe\Assets
+ */
+class ContentFileMigrationTask extends BuildTask
+{
+    /**
+     * @var string
+     */
+    protected $title = 'Content File Migration Task';
+
+    /**
+     * @var string
+     */
+    private static $segment = 'content-file-migration-task';
+
+    /**
+     * Mapping property for field updates
+     *
+     * @var array
+     */
+    private $mapping = [];
+
+    /**
+     * @param \SilverStripe\Control\HTTPRequest $request
+     */
+    public function run($request)
+    {
+        $this->setMapping();
+        $this->migrateContentFiles();
+    }
+
+    /**
+     * Migrate file image references to the new SS4 shortcode
+     */
+    protected function migrateContentFiles()
+    {
+        foreach ($this->getMapping() as $class => $tables) {
+            foreach ($tables as $table => $fields) {
+                $selectString = 'ID';
+
+                foreach ($fields as $field) {
+                    $selectString = "$selectString, \"{$field}\"";
+                }
+
+                $records = DB::query("SELECT {$selectString} FROM \"{$table}\"");
```

Comment:
This query should likely be run on the base, versions and live table (if available) for each class that has an eligible field.